pennylane/ops/functions/eigvals.py

```diff
@@ -93,6 +96,9 @@ def circuit(theta):
         )
         return qml.math.linalg.eigvalsh(qml.matrix(op))
 
+    if isinstance(op, scipy.sparse.coo_matrix):

```

Comment:
@soranjh note that `op` in this case would be `qml.SparseHamiltonian`! Not a sparse matrix directly.

So this should be something like

```python
if isinstance(op, qml.SparseHamiltonian) and k == ?:
    return scipy.sparse.linalg.eigsh(op.sparse_matrix(), k=?, which=?)
```

Another option? Maybe it makes sense to define the `compute_eigvals` method 

```python
class SparseHamiltonian(Observable):
    def __init__(self, H, k=?, which=?, wires=None, do_queue=True, id=None):
        if not isinstance(H, coo_matrix):
            raise TypeError("Observable must be a scipy sparse coo_matrix.")
        super().__init__(H, wires=wires, do_queue=do_queue, id=id)
        self.hyperparameters["k"] = k
        self.hyperparameters["which"] = which

    def compute_eigvals(H, k=?, which=?):
        return scipy.sparse.linalg.eigsh(H, k=?, which=?)
```

Reply:
Thanks Josh. I used the first option for consistency with a non-sparse Hamiltonian. Is there any specific benefit when using the second option?

The UI for the first option will be:
```
H = qml.hf.generate_hamiltonian(mol)(geometry)
Hs = qml.utils.sparse_hamiltonian(H)
Hs = qml.SparseHamiltonian(Hs, wires=all)
qml.eigvals(Hs)
```